### PR TITLE
Support renaming rooms

### PR DESCRIPTION
**Pull request by TheJetOU**
Client PR: smogon/pokemon-showdown-client#1447.

### CODE REVIEW

File: lib/fs.ts [hunk 1]
```diff
@@ -80,6 +80,18 @@ class FSPath {
 		return fs.readFileSync(this.path, options) as Buffer;
 	}
 
+	checkIfExists(): Promise<boolean> {
```
Comment (Zarel) [inline lib/fs.ts]: No other function prepends `checkIf` to its name.

```suggestion
	exists(): Promise<boolean> {
```

Comment (Zarel) [reply lib/fs.ts]: Also do we need this? In theory, `isFile`, `isFolder`, etc should cover this use-case.

Comment (TheJetOU) [reply lib/fs.ts]: > Also do we need this? In theory, isFile, isFolder, etc should cover this use-case.

I guess I missed them when skimming through it, but `FS#exists` maybe be worth keeping in other cases where we cannot infer if the file is a file or directory.

Comment (Zarel) [reply lib/fs.ts]: This hasn't been changed yet...

**Comment by TheJetOU** [comment]
Bump.

**Comment by Zarel** [comment]
Looks good, I think.

**Comment by Zarel** [comment]
Shit, forgot to merge this for this restart...

**Pull request merged**